### PR TITLE
Wishlist

### PR DESCRIPTION
**Pull request by bossmonkey98**
WishList functionality added

### CODE REVIEW

**Review by arjun-computer-geek — APPROVED** [review]
Everything looks good. Good code of conduct. Good job @bossmonkey98

File: src/components/Card/Card.css [hunk 1]
```diff
@@ -0,0 +1,59 @@
+.vertical-card{
+    padding: 5px;
+    flex-grow: 1;
+    background-color: var(--secondary-color);
+    max-width: 210px;
```
Comment (rutu14) [inline src/components/Card/Card.css:5]: Suggestion: Can convert it to rems.

File: src/components/Card/Card.css [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+.rating{
+    display: flex;
+    justify-content: center;
+    align-items: center;
+}
+
+#star{
+    font-size:1rem !important;
```
Comment (rutu14) [inline src/components/Card/Card.css:16]: Suggestion: Avoid using important.

File: src/components/Card/Card.css [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    width:90%;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    justify-content: space-evenly;
+    color: var(--secondary-color);
+    margin: 5px;
+}
+.btn:hover{
+    background-color:rgba(0, 0, 0, 0.648);
```
Comment (rutu14) [inline src/components/Card/Card.css:58]: Suggestion: Can use variable for color.

**Review by rutu14 — APPROVED** [review]
It is good to merge.

File: src/components/Card/Card.css (hunk 2 above)
Comment (t007rushi) [inline src/components/Card/Card.css:16]: avoid IMP

File: src/components/Card/Card.css [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    flex-grow: 1;
+    background-color: var(--secondary-color);
+    max-width: 210px;
+    margin:30px 0 0 30px;
+}
+
+.rating{
+    display: flex;
+    justify-content: center;
+    align-items: center;
```
Comment (t007rushi) [inline src/components/Card/Card.css:12]: can make a utility out of it

File: src/components/Card/Card.css [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    font-size:1rem !important;
+    color: gold;
+}
+
+.vertical-card-content{
+    font-size: 14px;
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
```
Comment (t007rushi) [inline src/components/Card/Card.css:25]: same here

File: src/components/Card/Card.jsx [hunk 6]
```diff
@@ -0,0 +1,40 @@
+import React, { useState } from 'react'
+import { Favorite,FavoriteBorderOutlined, ShoppingCart, Star } from '@material-ui/icons'
+import './Card.css'
+import { useAuth } from '../../components/context/auth-context'
+import { useWishlist } from '../../components/context/wishlist-context'
```
Comment (t007rushi) [inline src/components/Card/Card.jsx:5]: barrel export/import

File: src/components/Card/Card.jsx [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+import { useNavigate } from 'react-router-dom'
+
+
+
+const Card = ({items}) => {
```
Comment (t007rushi) [inline src/components/Card/Card.jsx:10]: you can destructure your keys here only

File: src/pages/Wishlist/Wishlist.jsx [hunk 8]
```diff
@@ -1,9 +1,21 @@
...
+
+export  function Wishlist(){
+  const { wishlist } = useWishlist()
 
-export default function Wishlist(){
   return (
-    <div>
-      Wishlist
+    <div className='wishlist-Container' >
+      {wishlist.length ===0? <>
```
Comment (t007rushi) [inline src/pages/Wishlist/Wishlist.jsx:11]: nice check here

File: src/pages/auth/Login.jsx [hunk 9]
```diff
@@ -1,24 +1,39 @@
...
-import { Link} from 'react-router-dom'
+import { Link, useNavigate} from 'react-router-dom'
+import {useAuth} from '../../components/context/auth-context'
+import {LogInHandler} from '../../components/auth/Loghandler'
+
 
 export default function Login() {
+  const {user,setUser} =useAuth()
+  const [loguser,setLoguser] =useState({email:'',pass:''})
+  const guestUser ={
```
Comment (t007rushi) [inline src/pages/auth/Login.jsx:11]: good the you have guest login

File: src/pages/auth/Login.jsx [hunk 10]
```diff
@@ -1,24 +1,39 @@
...
-                 placeholder="Enter Password"/>
+                 placeholder="Enter Password" onChange={(e)=>setLoguser({...loguser,pass:e.target.value})}/>
                  <p style={{cursor:"context-menu"}}><input type ="radio"/>Remember me Forget Password?</p>
-             <button className="btn">Login</button>
-              No Account?<Link to='/Signup'>CreatOne</Link>
+             <button className="btn" type='submit'>
+                  Login
+              </button>
+              <button className='btn' onClick={(e)=>{e.preventDefault();LogInHandler(guestUser,setUser,navigator)}}>
+                Login as Guest {console.log(user)}
```
Comment (t007rushi) [inline src/pages/auth/Login.jsx:34]: remove console

**Review by t007rushi — APPROVED** [review]
Look for suggested ones , else looks fine to me

File: src/pages/auth/Login.jsx (hunk 10 above)
Comment (bossmonkey98) [reply src/pages/auth/Login.jsx:34]: okay

**Pull request merged**